app/src/main/java/com/hadiyarajesh/notex/notification/NotificationWorker.kt

```diff
@@ -0,0 +1,23 @@
+package com.hadiyarajesh.notex.notification
+
+import android.content.Context
+import androidx.core.app.NotificationCompat
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.hadiyarajesh.notex.utility.Constants
+
+class NotificationWorker(private val context: Context,
+private val workerParameters: WorkerParameters):CoroutineWorker(context,workerParameters) {
+    override suspend fun doWork(): Result {
+        val id=inputData.getInt(Constants.Notification_Service_key,0)
+        //val priority=inputData.getInt("Priority",2)
+        triggerNotification(id,NotificationCompat.PRIORITY_HIGH)
+        return Result.success()
+    }
```

Comment:
Why we are directly using CoroutineWorker for this?

As we know here that we need to do this periodically or at a specific time as well. we can use two different workers `PeriodicWorkRequestBuilder` and this one to make it more readable and maintainable.

read more here [Notification using Work Manager](https://proandroiddev.com/custom-notification-with-work-manager-for-android-41f10090a75e)